File: src/main/java/com/pawan/choure/java/service/TradeEnricher.java
package com.pawan.choure.java.service;

import com.pawan.choure.java.model.Trade;
import org.springframework.stereotype.Component;

@Component
public class TradeEnricher {

    public Trade getEnrichedTrade(Trade trade){
        trade.setPnl_spn(0000);
        trade.setSpn(0000);
        return trade;
    }
}

File: src/main/kotlin/com/pawan/choure/kotlin/service/TradeEnricher.kt
package com.pawan.choure.kotlin.service

import com.pawan.choure.kotlin.model.Trade
import org.springframework.stereotype.Component

@Component
class TradeEnricher {
    fun getEnrichedTrade(trade: Trade): Trade {
        trade.pnl_spn = 0
        trade.spn = 0
        return trade
    }
}